### PR TITLE
Add Totem Pokemon

### PR DESCRIPTION
**Pull request by KrisOnline**

### CODE REVIEW

**Comment by Zarel** [comment]
Sprites though

**Comment by KrisOnline** [comment]
>Sprites though

they use the same minisprites as their regular formes so that is manageable.

Their gif sprites are basically indifferent to the regular forms with the difference being larger dimensions, but i dont see a point in implementing those if we can just recycle the sprites.

**Comment by Zarel** [comment]
Yes but you have to write the code to recycle the sprites.

**Comment by Marty-D** [comment]
Please use `Mimikyu-Busted-Totem`; I told Kaphotics we'd just tack on `-Totem` for PS imports/exports and he's already added support for it.

Re: event data, they all come with 3 perfect IVs, and you don't need to include the Ability when only one exists in their dex data.

Re: learnsets, remove all preexisting event moves (Mimikyu-Totem has eight `7S0` moves right now), and egg moves since, while they can breed, nothing can evolve into them.

Re: dex data, remove prevos and evoLevels.

**Comment by KrisOnline** [comment]
done @Marty-D

**Comment by Marty-D** [comment]
@Zarel It seems you already have all the Totem dex sprites that appeared in SM uploaded, and you even used the same naming scheme as this PR, for example [Mimikyu-Busted-Totem](https://play.pokemonshowdown.com/sprites/xydex/mimikyu-busted-totem.png).
Do you want to use those and copy over the other animated sprites to have a complete collection or recycle sprites? Either way I'll be redirecting their minisprites to the appropriate location.

**Comment by Zarel** [comment]
I'd say code it into the engine (`battledata.js`) instead of straight copy/pasting them.

File: data/pokedex.js [hunk 1]
```diff
@@ -11719,6 +11747,21 @@ exports.BattlePokedex = {
 		prevo: "yungoos",
 		evoLevel: 20,
 		eggGroups: ["Field"],
+		otherFormes: ["gumshoostotem"],
```
Comment (DeathlyPlays) [inline data/pokedex.js]: What happened to [Pokemon]-Alola-Totem

Comment (Marty-D) [reply data/pokedex.js]: What are you talking about? Alolan Gumshoos is not a thing...

Comment (KrisOnline) [reply data/pokedex.js]: what

Comment (KrisOnline) [reply data/pokedex.js]: like you didnt even comment on anything relating to that you just commented to gumshoos's otherFormes line

Comment (DeathlyPlays) [reply data/pokedex.js]: I swear Im an idiot

**Comment by TheImmortal** [comment]
Also, ``"pokeball": "pokeball"`` should be omitted. I see @Marty-D included it in some recent events but since the beginning I've omitted it. In other words, no Pokeball means the default ball aka ``pokeball``.

**Comment by Marty-D** [comment]
I've been using `"pokeball": "pokeball"` to differentiate events and gifts that only come in a Poké Ball from things you can actually catch in the game with any legal Ball (which stay blank). It would be a pretty bad event database if that kind of information were left out.

**Comment by Zarel** [comment]
It certainly is possible to hardcode that no `pokeball` field means it's a pokeball. Would make the event database easier to read. I'd be fine with either, but having the `pokeball` definitely does help with certain things.

**Comment by TheImmortal** [comment]
In that case @Marty-D there probably are a ton of past gen events missing said Pokeball. Off course this was before we started including in-game stuff.

**Pull request merged**